scripts/workflow/bootstrapper.sh

```diff
@@ -56,40 +60,141 @@ if [ $EXPORTS_SET = 1 ] ; then
+# the directory that holds the stack to terraform
+STACK_DIR=$PWD/$TERRAFORM_DIR/$STACK
+# remove any previous local backend for stack
+rm -rf "$STACK_DIR"/.terraform
+rm -f "$STACK_DIR"/.terraform.lock.hcl
+#  copy shared tf files to stack
+cp "$ROOT_DIR"/"$INFRASTRUCTURE_DIR"/common/locals.tf "$STACK_DIR"
+cp "$ROOT_DIR"/"$INFRASTRUCTURE_DIR"/common/provider.tf "$STACK_DIR"
+cp "$ROOT_DIR"/"$INFRASTRUCTURE_DIR"/common/common-variables.tf "$STACK_DIR"
+
+if [[ "$USE_REMOTE_STATE_STORE" =~ ^(true|yes|y|on|1|TRUE|YES|Y|ON) ]]; then
+  cp "$ROOT_DIR"/"$INFRASTRUCTURE_DIR"/remote/versions.tf "$STACK_DIR"
+else
+  cp "$ROOT_DIR"/"$INFRASTRUCTURE_DIR"/local/versions.tf "$STACK_DIR"
+fi
+# switch to target stack directory ahead of tf init/plan/apply
+cd "$STACK_DIR" || exit
+# if no stack tfvars create temporary one
+TEMP_STACK_TF_VARS_FILE=0
+if [ ! -f "$ROOT_DIR/$INFRASTRUCTURE_DIR/$STACK_TF_VARS_FILE" ] ; then
+  touch "$ROOT_DIR/$INFRASTRUCTURE_DIR/$STACK_TF_VARS_FILE"
+  TEMP_STACK_TF_VARS_FILE=1
 fi
 
-# deploy stack
-/bin/bash ./scripts/v2/action-infra-stack.sh
+# init terraform
+terraform-initialise
+
+if [ -n "$ACTION" ] && [ "$ACTION" = 'plan' ] ; then
+  terraform plan \
+  -var-file $ROOT_DIR/$INFRASTRUCTURE_DIR/$COMMON_TF_VARS_FILE \
+  -var-file $ROOT_DIR/$INFRASTRUCTURE_DIR/$STACK_TF_VARS_FILE \
+  -var-file $ROOT_DIR/$INFRASTRUCTURE_DIR/$PROJECT_TF_VARS_FILE \
+  -var-file $ROOT_DIR/$INFRASTRUCTURE_DIR/$ENV_TF_VARS_FILE
+fi
+if [ -n "$ACTION" ] && [ "$ACTION" = 'apply' ] ; then
+  terraform apply -auto-approve \
+  -var-file $ROOT_DIR/$INFRASTRUCTURE_DIR/$COMMON_TF_VARS_FILE \
+  -var-file $ROOT_DIR/$INFRASTRUCTURE_DIR/$STACK_TF_VARS_FILE \
+  -var-file $ROOT_DIR/$INFRASTRUCTURE_DIR/$PROJECT_TF_VARS_FILE \
+  -var-file $ROOT_DIR/$INFRASTRUCTURE_DIR/$ENV_TF_VARS_FILE
+fi
+if [ -n "$ACTION" ] && [ "$ACTION" = 'destroy' ] ; then
+  terraform destroy -auto-approve \
+  -var-file $ROOT_DIR/$INFRASTRUCTURE_DIR/$COMMON_TF_VARS_FILE \
+  -var-file $ROOT_DIR/$INFRASTRUCTURE_DIR/$STACK_TF_VARS_FILE \
+  -var-file $ROOT_DIR/$INFRASTRUCTURE_DIR/$PROJECT_TF_VARS_FILE \
+  -var-file $ROOT_DIR/$INFRASTRUCTURE_DIR/$ENV_TF_VARS_FILE
+fi
+# cleardown temp files
+rm -f "$STACK_DIR"/common-variables.tf
+rm -f "$STACK_DIR"/locals.tf
+rm -f "$STACK_DIR"/provider.tf
+rm -f "$STACK_DIR"/versions.tf
+
+if [[ $TEMP_STACK_TF_VARS_FILE == 1 ]]; then
+  rm "$ROOT_DIR/$INFRASTRUCTURE_DIR/$STACK_TF_VARS_FILE"
+fi
+
+# back to root
+cd "$ROOT_DIR" || exit
+
+# having build the stack using a local backend we need to migrate the state held locally to newly build remote
 
-# having build the stack using a local backend we need to push
-# the state to the remote
 if ! $USE_REMOTE_STATE_STORE  ; then
   # check if remote state bucket exists we are okay to migrate state to it
-  if aws s3api head-bucket --bucket "$TERRAFORM_BUCKET_NAME" 2>/dev/null; then
+  if aws s3api head-bucket --bucket "$$TF_VAR_terraform_state_bucket_name" 2>/dev/null; then
```

Comment:
You've got a double $ here. Might not cause probs, but might do

Reply:
Good spot - thanks